eureka-core/src/main/java/com/netflix/eureka/util/AwsAsgUtil.java

```diff
@@ -195,6 +211,62 @@ private AutoScalingGroup retrieveAutoScalingGroup(String asgName) {
         }
     }
 
+    private Credentials initializeStsSession(String asgAccount) {
+        AWSSecurityTokenService sts = new AWSSecurityTokenServiceClient(new InstanceProfileCredentialsProvider());
+        String region = DiscoveryManager.getInstance().getEurekaClientConfig()
+                .getRegion();
+        if (!region.equals("us-east-1")) {
+            sts.setEndpoint("sts." + region + ".amazonaws.com");
+        }
+
+        String roleArn = "arn:aws:iam::" + asgAccount + ":role/ListAutoScalingGroups";
```

Comment:
Can you make the role name (ListAutoScalingGroups) a configurable parameter? You can add it as a config param to EurekaServerConfig/DefaultEurekaServerConfig with this as the default.